src/main/java/br/com/metricminer2/scm/GitRepository.java

```diff
@@ -410,4 +422,21 @@ public String blame(String file, String commitToBeBlamed, Integer line) {
 		}
 	}
 
+	public int getMaxNumberFilesInACommit() {
+		if(this.maxNumberFilesInACommit == null){
+			return DEFAULT_MAX_NUMBER_OF_FILES_IN_A_COMMIT;
+		} else {
+			return this.maxNumberFilesInACommit;
+		}
+	}
+
+	@Override
+	public void setMaxNumberFilesInACommit(int maxNumber) {
```

Comment:
Both options would be initialization parameters, indeed. However, the masterBranch might not be known at construction time. That's the intent of `git.getRepository().getBranch()` when opening the repository. Thus, is that OK to leave masterBranch as non-constructor parameter?

I'm adding maxNumber to configure GitRepository, GitRemoteRepository and SubversionRepository by consctructor and static factory methods using fluent interface.

Only GitRemoteRepository need a little more work, since it will have 1 mandatory parameter (url) and 3 optional (temporaryDir, bareRepos and maxNumber). Then I'm trying to implement a separeted builder.